Removal of raw Scene and View constructors from public API

LinusVanElswijk opened the pull request:
This is a rework of PR #337.
Refactor of Scene and View class code to make it easier to read and maintain:

- Replaced index based loops, with foreach loops where possible
- Removed raw constructors from public API
- Removed commented out code
- Removed unused control flow parameters of internal functions
- Replaced pointer parameters by reference parameters where nullptr would be illegal as argument.

No functional changes.

simonfuhrmann commented:
Can you please provide smaller PRs, ideally with one commit each? This is really hard to review, and smaller PRs have shorter round-trips.

LinusVanElswijk commented:
No problem, I'll work on that.

LinusVanElswijk commented:
Update: Changes are now restricted to the removal of unsafe constructors and factory functions from the public API of View and Scene:

- Removed raw constructors from View and Scene class API.
- Removed Scene::Create(void), which returns scenes in a weird and undocumented state. Directly initializing with Scene::Create(std::string) should be preferred and pointing to null should be preferred over  pointing to partially initialized scenes.

LinusVanElswijk commented on tests/mve/gtest_view.cc:
```diff
@@ -8,176 +8,179 @@
 
```
The only real change in this file is the addition on line 12. Everything else is just replacement of a View with View::Ptr.

simonfuhrmann commented on tests/mve/gtest_view.cc:
```diff
@@ -8,176 +8,179 @@
-    EXPECT_TRUE(view3.is_dirty());
-    EXPECT_TRUE(view3.get_meta_data().is_dirty);
+    mve::View::Ptr view1 = mve::View::create(),
+                   view2 = mve::View::create(),
```
Please make three instructions for this.

simonfuhrmann commented on libs/mve/scene.h:
```diff
@@ -95,6 +93,7 @@ class Scene
     bool bundle_dirty;
 
 private:
+    Scene (void);
```
It's customary to make this protected if the class is not final.

LinusVanElswijk commented on libs/mve/scene.cc:
```diff
@@ -120,19 +96,18 @@ Scene::get_total_mem_usage (void)
-        if (this->views[i] != nullptr)
-            ret += this->views[i]->get_byte_size();
-    return ret;
+    return std::accumulate(views.begin(), views.end(), std::size_t{0},
```
It was not a conscient decision.
I got into the habit of using list initialization, because this form forbids narrowing conversions.
